src/platform/linux_usbfs/device.rs

```diff
@@ -54,8 +56,105 @@ impl LinuxDevice {
         let fd = rustix::fs::open(&path, OFlags::RDWR | OFlags::CLOEXEC, Mode::empty())
             .inspect_err(|e| warn!("Failed to open device {path:?}: {e}"))?;
 
+        Self::create_inner(fd, Some(d.path.clone()), active_config, busnum, devnum)
+    }
+
+    pub(crate) fn from_fd(fd: OwnedFd) -> Result<Arc<LinuxDevice>, Error> {
+        debug!("Wrapping fd {} as usbfs device", fd.as_raw_fd());
+
+        // NOTE: must be called from the same thread that gave us the FD on android, otherwise this fails
+        let proc_path = PathBuf::from(format!("/proc/self/fd/{}", fd.as_raw_fd()));
+
+        debug!("Reading link: {proc_path:?}");
+        let fd_path = std::fs::read_link(&proc_path)?;
+
+        // TODO: should we switch to regex or regex-lite for this?
+        let prefix = "/dev/bus/usb/";
+        let Ok(dev_num_bus_num) = fd_path.strip_prefix(prefix) else {
+            error!(
+                "Usb device path does not start with required prefix `{prefix}`: {}",
+                fd_path.to_string_lossy()
+            );
+            return Err(ErrorKind::Other.into());
+        };
+        let Some(dev_num_bus_num) = dev_num_bus_num.to_str() else {
+            error!("End of usb device path is not UTF-8!: {dev_num_bus_num:?}",);
+            return Err(ErrorKind::Other.into());
+        };
+
+        let mut split = dev_num_bus_num.split("/");
+        let Some(bus_str) = split.next() else {
+            error!("Failed to split usbfs device path: {}", dev_num_bus_num);
+            return Err(ErrorKind::Other.into());
+        };
+        let Some(dev_str) = split.next() else {
+            error!("Failed to split usbfs device path: {}", dev_num_bus_num);
+            return Err(ErrorKind::Other.into());
+        };
+
+        let Ok(busnum) = bus_str.parse::<u8>() else {
+            error!("Usb bus number failed to parse: {bus_str}",);
+            return Err(ErrorKind::Other.into());
+        };
+        let Ok(devnum) = dev_str.parse::<u8>() else {
+            error!("Usb device number failed to parse: {dev_str}",);
+            return Err(ErrorKind::Other.into());
+        };
+
+        const LIBUSB_REQUEST_GET_CONFIGURATION: u8 = 0x08;
```

Comment:
```suggestion
        const REQUEST_GET_CONFIGURATION: u8 = 0x08;
```

(it's a request defined by the USB standard, not something from libusb)